### PR TITLE
DOCSP-41879 Clarify SaveChanges is transactional

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

This PR also adds a few lines of `EnsureCreated()` where it was missed in a previous PR

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-41879
Staging - https://preview-mongodbjordansmith721.gatsbyjs.io/entity-framework/DOCSP-41879-save-changes-transactional/fundamentals/write-data/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/write-data.txt [hunk 1]
```diff
@@ -23,20 +23,32 @@ Overview
...
+When you call the ``SaveChanges()`` or ``SaveChangesAsync()`` method, the {+provider-short+} automatically detects
 any changes made to your data and runs the necessary commands to update the
 database by using the MongoDB Query API.
 
-In this guide, you can see examples of how to perform common write operations on
-an application configured to use the {+provider-short+}.
+The ``SaveChanges()`` and ``SaveChangesAsync()`` methods are transactional by
+default. This means that if an error occurs during an operation, the provider
+rolls back any changes made to the database. Because of this, your application
+must be connected to a transaction-capable instance of a MongoDB server, such as
```
Comment (mongoKart) [inline source/fundamentals/write-data.txt:35]: i believe this is the preferred wording

```suggestion
must be connected to a transaction-capable deployment of MongoDB Server, such as
```

File: source/fundamentals/write-data.txt [hunk 2]
```diff
@@ -23,20 +23,32 @@ Overview
...
+a replica set. You can disable transactions during write operations by setting
```
Comment (mongoKart) [inline source/fundamentals/write-data.txt:36]: S: Make this a new paragraph

```suggestion
a replica set.

You can disable transactions during write operations by setting
```

**Review by mongoKart — APPROVED** [review]
nice. lgtm w/ a couple suggestions

File: source/fundamentals/write-data.txt [hunk 3]
```diff
@@ -23,20 +23,34 @@ Overview
...
 
-In this guide, you can see examples of how to perform common write operations on
-an application configured to use the {+provider-short+}.
+The ``SaveChanges()`` and ``SaveChangesAsync()`` methods are transactional by
+default. This means that if an error occurs during an operation, the provider
+rolls back any changes made to the database. Because of this, your application
+must be connected to a transaction-capable deployment of MongoDB server, such as
+a replica set.
+
+You can disable transactions during write operations by setting
```
Comment (damieng) [inline source/fundamentals/write-data.txt:38]: I'd probably clarify this "You can disable these automatic transactions within SaveChanges and SaveChangeaAsync by setting..." as other transaction facilities in EF exist which we don't yet support (explicit transactions).

We should clarify that this is not recommended as any kind of concurrency change, database operation failure, duplicate key etc. will leave the database with only some of the changes applied.

File: source/fundamentals/write-data.txt [hunk 4]
```diff
@@ -37,13 +37,35 @@ an application configured to use the {+provider-short+}.
...
+-------------------------------
+
+The ``SaveChanges()`` and ``SaveChangesAsync()`` methods are transactional by
+default. This means that if an error occurs during an operation, the provider
+rolls back any changes made to the database. Because of this, your application
+must be connected to a transaction-capable deployment of MongoDB server, such as
+a replica set.
+
+You can disable automatic transactions in ``SaveChanges()`` and
+``SaveChangesAsync()`` by setting the ``AutoTransactionBehavior`` property to
```
Comment (mongoKart) [inline source/fundamentals/write-data.txt:50]:
```suggestion
You can disable automatic transactions in the ``SaveChanges()`` and
``SaveChangesAsync()`` methods by setting the ``AutoTransactionBehavior`` property to
```

**Review by mongoKart — APPROVED** [review]
1 suggestion

File: source/fundamentals/write-data.txt (hunk 3 above)
Comment (jordan-smith721) [reply source/fundamentals/write-data.txt]: This was getting a little long and involved so I moved it to it's own section just below the overview and addressed these points there.

File: source/fundamentals/write-data.txt [hunk 5]
```diff
@@ -37,13 +37,35 @@ an application configured to use the {+provider-short+}.
...
+a replica set.
+
+You can disable automatic transactions in the ``SaveChanges()`` and
+``SaveChangesAsync()`` methods by setting the ``AutoTransactionBehavior`` property to
+``AutoTransaction.Never`` on your ``DbContext`` subclass during application
+setup. However, we do not recommend disabling this feature. Doing so causes
+any concurrency changes or operation failures during the save operation
+to leave the database in an inconsistent state.
+
+The following example shows how to disable automatic transactions in the
```
Comment (damieng) [inline source/fundamentals/write-data.txt:56]: Maybe prefix with "If, however, you choose to do this" for people speed-reading/scanning next to not just land here and think the answer is to disable transactions? Hopefully will send them back up to the previous paragraph.

Comment (jordan-smith721) [reply source/fundamentals/write-data.txt:56]: Added a warning for emphasis as this aligns more with our typical style

**Pull request merged**